### PR TITLE
Add ability to manage custom DNS records

### PR DESCRIPTION
**Pull request by Mograine**
**By submitting this pull request, I confirm the following:** `{please fill any appropriate checkboxes, e.g: [X]}`

`{Please ensure that your pull request is for the 'devel' branch!}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [ ] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [X] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff))

---

**What does this PR aim to accomplish?:**

This new page will take advantage of the two commands added in core PR https://github.com/pi-hole/pi-hole/pull/2978 to allow users to add and delete custom DNS records

This intend to correctly fix core issue https://github.com/pi-hole/pi-hole/issues/975

**How does this PR accomplish the above?:**

This interface use datatable to manage the custom records, using the two new commands in order to add/delete records from the custom.list file

**What documentation changes (if any) are needed to support this PR?:**

Probably a bunch of screenshot to update since this PR add a new navbar entry.

### CODE REVIEW

File: scripts/pi-hole/php/customdns.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+            if (empty($domain))
+                return errorJsonResponse("Domain must be set");
+
+            if (!is_valid_domain_name($domain))
+                return errorJsonResponse("Domain must be valid");
+
+            $existingEntries = getCustomDNSEntries();
+
+            if (array_key_exists($domain, $existingEntries))
```
Comment (flo-wer) [inline scripts/pi-hole/php/customdns.php]: This should take care of the case someone wants to add multiple IP addresses for the same domain, e.g. separate IPv4 and IPv6 address.

**Comment by PromoFaux** [comment]
The core functionality of this appears to be working, however is it intentional that the errors are just a plain js `alert()`? See the white/blacklist page for example of how outputs are displayed there.

There also does not seem to be any positive feedback that the request is being processed. For example, if I click add.. nothing happens for a few seconds, then suddenly the entry pops up in the table below.

Could we also please amend the table so that there is no footer row?

Other than that, great stuff!

**Comment by Mograine** [comment]
@PromoFaux Should be good now, more feedback on both projects :)

**Comment by PromoFaux** [comment]
Great stuff, will try and find some time to review tomorrow. Cheers!

**Review by PromoFaux — APPROVED** [review]
Thanks, that fits much better with the other similar pages now.

Tested and confirmed to be working

**Pull request merged**